src/App.tsx

```diff
@@ -1,19 +1,153 @@
 import React from 'react';
-import './App.css';
-
+import './App.scss';
+import todos from './api/todos';
 import users from './api/users';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+const todosCopy = todos.map(todo => ({ ...todo }));
+const prepareTodos = () => {
+  return todosCopy.map(todo => ({
+    ...todo,
+    user: users.find(user => user.id === todo.userId),
+  }));
+};
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todos: Todo[],
+  employeeId: number,
+  task: string,
+  errorEmployee: boolean,
+  errorTask: boolean,
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    todos: prepareTodos(),
+    employeeId: 0,
+    task: '',
+    errorEmployee: false,
+    errorTask: false,
+  };
+
+  handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    this.setState(prevState => {
+      if (!prevState.employeeId) {
+        return {
+          ...prevState,
+          errorEmployee: true,
+        };
+      }
+
+      if (!prevState.task) {
+        return {
+          ...prevState,
+          errorTask: true,
+        };
+      }
+
+      if (!prevState.employeeId && !prevState.task) {
+        return {
+          ...prevState,
+          errorEmployee: true,
+          errorTask: true,
+        };
+      }
+
+      const maxId = Math.max(...prevState.todos.map(todo => todo.id));
+
+      todosCopy.push({
+        id: maxId + 1,
+        title: prevState.task,
+        userId: prevState.employeeId,
+        completed: false,
+      });
```

Comment:
why do you push to todosCopy? remove this
you need to update todos inside state